cachi2/core/models/property_semantics.py

```diff
@@ -101,3 +101,100 @@ def merge(self, other: "Self") -> "Self":
             pip_package_binary=self.pip_package_binary or other.pip_package_binary,
             bundler_package_binary=self.bundler_package_binary or other.bundler_package_binary,
         )
+
+
+def merge_relationships(
```

Comment:
The problem of this definition is that it puts the burden of understanding how this function works on its client. Can any of these lists be empty? Can they be of different lengths? `doc_ids` is a list of strings, would it be ok if someone just passed `["foo", "bar", "baz"]` as `doc_ids`? Judging from the code this function could accept just a list of sboms to merge and extract all necessary data by itself.